lib/core/src/rcMisc.cpp

```diff
@@ -4608,12 +4651,27 @@ int load_in_addr_from_hostname(const char* _hostname, struct in_addr* _out) {
     if (ret_getaddrinfo_with_retry) {
         return ret_getaddrinfo_with_retry;
     }
+
     *_out = reinterpret_cast<struct sockaddr_in*>(p_addrinfo->ai_addr)->sin_addr;
+
+    if (CLIENT_PT != ::ProcessType) {
+        rodsLog(LOG_DEBUG, "%s :: Caching addrinfo for [hostname=%s].", __func__, _hostname);
+
+        //try {
+            const auto age = irods::get_server_property<std::chrono::seconds>(irods::CFG_EVICTION_AGE_IN_SECONDS_KW);
+            net::dnsc_insert_or_assign(target, *p_addrinfo, age);
+        //}
+        //catch (...) {
+            // TODO Should the default be specified in the catch statement.
+            //net::dnsc_insert_or_assign(target, *p_addrinfo, std::chrono::seconds{30});
+        //}
```

Comment:
the outcome of these lines depend on how we want to set the eviction property. see rodsServer.cpp.